datafusion/physical-expr/src/array_expressions.rs

```diff
@@ -1515,32 +1515,33 @@ pub fn array_union(args: &[ArrayRef]) -> Result<ArrayRef> {
     }
     let array1 = &args[0];
     let array2 = &args[1];
+
+    fn union_arrays<O: OffsetSizeTrait>(
+        array1: &ArrayRef,
+        array2: &ArrayRef,
+        l_field_ref: &Arc<Field>,
+        r_field_ref: &Arc<Field>,
+    ) -> Result<ArrayRef> {
+        match (l_field_ref.data_type(), r_field_ref.data_type()) {
+            (DataType::Null, _) => Ok(array2.clone()),
```

Comment:
Awesome @Weijun-H 
Makes the code cleaner

Its not a problem of this PR but I'm thinking why we need to clone arrays in case of nulls....  🤔

Reply:
If array1 is Null, return array2 directly because it is a union function. @comphead